pkg/framework/test/etcd.go

```diff
@@ -12,10 +13,10 @@ import (
 
 // Etcd knows how to run an etcd server. Set it up with the path to a precompiled binary.
 type Etcd struct {
-	// The path to the etcd binary
 	Path           string
 	EtcdURL        string
-	session        *gexec.Session
+	ProcessStarter simpleSessionStarter
```

Comment:
These changes seem to always be the same. How much more complex would it be to share some of this code between `apiserver` and `etcd`. And I can forsee other components in the future, will you have to duplicate this code again and again?

Reply:
That is indeed true. I can check tomorrow if we can unify them.